src/bootstrap/cache.rs

```diff
@@ -114,43 +113,49 @@ impl Deref for Interned<String> {
     type Target = str;
     fn deref(&self) -> &'static str {
         let l = INTERNER.strs.lock().unwrap();
-        unsafe { mem::transmute::<&str, &'static str>(l.get(*self)) }
+        let s: &str = l.get(*self).as_ref();
```

Comment:
What's `as_ref` called on? If it's `&String` then you should remove `.as_ref()` and let it coerce.

Reply:
Same for all the other 5 cases in this file.